benchmark/url/whatwg-url-to-and-from-path.js

```diff
@@ -3,28 +3,50 @@ const common = require('../common.js');
 const { fileURLToPath, pathToFileURL } = require('node:url');
 const isWindows = process.platform === 'win32';
 
-const bench = common.createBenchmark(main, {
-  input: isWindows ? [
-    'file:///c/',
-  ] : [
-    'file:///dev/null',
-    'file:///dev/null?key=param&bool',
-    'file:///dev/null?key=param&bool#hash',
-  ],
-  method: isWindows ? [
-    'fileURLToPath',
-  ] : [
-    'fileURLToPath',
-    'pathToFileURL',
-  ],
-  n: [5e6],
-});
+const inputs = isWindows ? [
+  'C:\\foo',
+  'C:\\Program Files\\Music\\Web Sys\\main.html?REQUEST=RADIO',
+  '\\\\nas\\My Docs\\File.doc',
+  'file:///C:/foo',
+  'file:///C:/dir/foo?query=1',
+  'file:///C:/dir/foo#fragment',
+] : [
+  '/dev/null',
+  '/dev/null?key=param&bool',
+  '/dev/null?key=param&bool#hash',
+  'file:///dev/null',
+  'file:///dev/null?key=param&bool',
+  'file:///dev/null?key=param&bool#hash',
+];
 
-function main({ n, input, method }) {
-  method = method === 'fileURLOrPath' ? fileURLToPath : pathToFileURL;
+const bench = common.createBenchmark(
+  main,
+  {
+    method: ['pathToFileURL', 'fileURLToPath'],
+    input: Object.values(inputs),
+    n: [5e6],
+  },
+  {
+    combinationFilter: (p) => {
+      if (!isWindows) {
+        return (
+          (p.method === 'pathToFileURL') ||
+          (p.input.startsWith('file://') && p.method === 'fileURLToPath')
+        );
+      }
+      return (
+        (!p.input.startsWith('file://') && p.method === 'pathToFileURL') ||
+        (p.input.startsWith('file://') && (!isWindows || p.method === 'fileURLToPath'))
+      );
```

Comment:
Can we avoid duplicating the conditions?
```suggestion
    combinationFilter: (p) => (
       (isWindows ?
         (!p.input.startsWith('file://') && p.method === 'pathToFileURL') :
         p.method === 'pathToFileURL'
       ) ||
       (p.input.startsWith('file://') && p.method === 'fileURLToPath')
    ),
```

or, maybe we write a `switch` instead
```suggestion
    combinationFilter: (p) => {
      switch (p.method) {
        case 'pathToFileURL': {
          // `:` is not a valid char for a filename on Windows (it is elsewhere)
          return isWindows ? !p.input.includes(':') : true;
        }
        
        case 'fileURLToPath':
          return p.input.startsWith('file://');
          
        default:
          throw 'unreachable';
      }
    },
```

Reply:
I really appreciate your thorough reviews. I completely agree that duplicating conditions isn't ideal, but I struggled to find a way to avoid duplication while ensuring that others who read my code later can still understand it easily. I've updated the code according to your suggestions, added the "long" UNC case, and avoided duplication.